testsuite/MDAnalysisTests/core/test_groups.py

```diff
@@ -603,6 +603,44 @@ def test_groupby_int(self, u):
         for g in gb.values():
             assert len(g) == 5
 
+    # tests for multiple attributes as arguments
+
+    def test_groupby_float_string(self, u):
+        gb = u.atoms.groupby(['charges', 'segids'])
+
+        for ref in [-1.5, -0.5, 0.0, 0.5, 1.5]:
+            for subref in ['SegA','SegB','SegC','SegD','SegE']:
+                assert (ref, subref) in gb.keys()
+                a = gb[(ref, subref)]
+                assert len(a) == 5
+                assert all(a.charges == ref)
+                assert all(a.segids  == subref)
+
+    def test_groupby_int_float(self, u):
+        gb = u.atoms.groupby(['resids', 'charges'])
+
+        uplim=len(gb)/5+1
```

Comment:
This value will be a `float` in python3. This is because any division in python3 produces a float. You will need an explicit int cast.